controllers/controllers/workloads/integration/cforg_controller_integration_test.go

```diff
@@ -0,0 +1,213 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"code.cloudfoundry.org/cf-k8s-controllers/api/repositories"
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/workloads/v1alpha1"
+	"code.cloudfoundry.org/cf-k8s-controllers/controllers/controllers/workloads"
+	. "code.cloudfoundry.org/cf-k8s-controllers/controllers/controllers/workloads/testutils"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	hncv1alpha2 "sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+var _ = Describe("CFOrgReconciler Integration Tests", func() {
+	const (
+		orgName = "my-org"
+	)
+	var (
+		testCtx       context.Context
+		rootNamespace *v1.Namespace
+		orgGUID       string
+		cfOrg         workloadsv1alpha1.CFOrg
+	)
+
+	waitForAnchor := func(anchorNamespace string, anchorLabel map[string]string, done chan bool) (*hncv1alpha2.SubnamespaceAnchor, error) {
+		for {
+			select {
+			case <-done:
+				return nil, fmt.Errorf("waitForAnchor was 'signalled' to stop polling")
+			default:
+			}
+
+			var subspaceAnchorList hncv1alpha2.SubnamespaceAnchorList
+			err := k8sClient.List(testCtx, &subspaceAnchorList, client.InNamespace(anchorNamespace), client.MatchingLabels(anchorLabel))
+			if err != nil {
+				return nil, fmt.Errorf("waitForAnchor failed")
+			}
+			if len(subspaceAnchorList.Items) > 1 {
+				return nil, fmt.Errorf("waitForAnchor found multiple anchors")
+			}
+			if len(subspaceAnchorList.Items) == 1 {
+				return &subspaceAnchorList.Items[0], nil
+			}
+
+			time.Sleep(time.Millisecond * 100)
+		}
+	}
+
+	// simulateHNC waits for the subnamespaceanchor to appear, and then
+	// creates the namespace, creates the hierarchyconfiguration in the
+	// namespace and sets the status on the subnamespaceanchor to Ok. It
+	// will stop waiting for the subnamespaceanchor to appear if anything
+	// is written to the done channel
+	simulateHNC := func(anchorNamespace string, anchorLabel map[string]string, done chan bool) {
```

Comment:
Running the HNC simulation function in the background is trying to fill the gap between envtests and a real cluster with HNC deployed. A "running HNC" is not needed for all controllers tests, for example it is not needed to verify the attributes of the subnamespace anchor itself.

Tests that check the controller behaviour when the namespace is there could be moved into a nested context (e.g. `When("the underlying namespace appears"....`) that creates the namespace in its `BeforeEach`. This would significantly simplify the code and there would be no need for switches like `doHNCSimulation`, or the `done` channel.

Reply:
Removed the HNC simulation and manually set the status to OK for the appropriate test cases. This greatly simplified the integration test.